### PR TITLE
145 L2 grid system

### LINKED ISSUE #145: L2 Grid System

### PR DESCRIPTION
**Pull request by saurabh-khare**
The PR brings the layout for L2 pages
1. Side navigation auto populated on L2 pages if the template of the page is hub-l2
2. Side navigation will be populated on the L2 page if it is in the root directory of a hub or if `sideNav` is explicitly set to the navigation document (in both cases sidenav.docx must be present)
3. Side navigation moved to its own block (previously was in header block) to avoid CLS and sending unnecessary bytes to L3/L1 pages. 
4. New template introduced for L2 pages
5. The template injects the side navigation and containerize hero block (as like L3 pages). 

Fix #145 

Test URLs:
- Before: https://main--hlx-test--urfuwo.hlx.page/blog
- After: https://145-l2-grid-system--hlx-test--urfuwo.hlx.live/blog

- [x] New Blocks introduced in this PR
      If yes, please provide details below

Block Name    | Documentation | Library Link
------------- | -------------|----------------
side-nav | [doc-link](https://sap.sharepoint.com/:w:/r/sites/207899/Shared%20Documents/website/blog/nav.docx?d=w4ebf77458b58438e93777e54f8b617c2&csf=1&web=1&e=y59QWP)

### CODE REVIEW

**Comment by saurabh-khare** [comment]
Can we merge this soon as it as lot of changes and will require continuous updates everytime main is updated

**Review by pahupe — CHANGES_REQUESTED** [review]
Body (central) column: the differentiation of “wide rows” and “narrow rows” is missing
- any “default” section on an L2 page (featured, latest, etc.) is a “narrow row”, i.e., there should be some spacing to the left of the body narrow row between the side nav narrow and the narrow row (which is there)
- but the hero on an L2 page is a “wide row”, i.e., there should be no spacing to the left of the hero between the side nav and the hero

But, when you look at the hero (might help to change its background-color to green) and the next elements (Featured, Latest, …), the hero should start ‘further left’ than the others. See as a reference: git branch grid-validation file sketches/l2grid.html

**Comment by pahupe** [comment]
- Validated L2 page (https://145-l2-grid-system--hlx-test--urfuwo.hlx.live/blog/) including background-colored hero) section styles. Looks all fine in all breakpoints
- Validated L3 pages for potential spill-overs. Looks good to me.

LGTM

File: scripts/aem.js [hunk 1]
```diff
@@ -658,6 +658,18 @@ async function loadFooter(footer) {
   return loadBlock(footerBlock);
 }
 
+/**
+ * Loads a block named 'footer' into footer
```
Comment (pahupe) [inline scripts/aem.js:662]: Should be 'sidenav', not footer

**Review by rootpea — APPROVED** [review]

**Pull request merged**